Neos.Media/Classes/Command/MediaCommandController.php

```diff
@@ -338,15 +367,87 @@ public function renderThumbnailsCommand(int $limit = null, bool $quiet = false)
                 break;
             }
         }
-        !$quiet && $this->output->progressFinish();
+    }
+
+    /**
+     * Render asset variants
+     *
+     * Loops over missing configured asset variants and renders them. Optional ``limit`` parameter to
+     * limit the amount of variants to be rendered to avoid memory exhaustion.
+     *
+     * If the re-render parameter is given, any existing variants will be rendered again, too.
+     *
+     * @param integer $limit Limit the amount of variants to be rendered to avoid memory exhaustion
+     * @param bool $quiet If set, only errors will be displayed.
+     * @param bool $recreate
+     * @return void
+     * @throws AssetVariantGeneratorException
+     * @throws IllegalObjectTypeException
+     */
+    public function renderVariantsCommand($limit = null, bool $quiet = false, bool $recreate = false): void
+    {
+        $resultMessage = null;
+        $generatedVariants = 0;
+        $configuredVariantsCount = 0;
+        $configuredPresets = $this->assetVariantGenerator->getVariantPresets();
+        foreach ($configuredPresets as $configuredPreset) {
+            $configuredVariantsCount += count($configuredPreset->variants());
+        }
+
+        $classNames = $this->reflectionService->getAllImplementationClassNamesForInterface(VariantSupportInterface::class);
+        foreach ($classNames as $className) {
+            /** @var AssetRepository $repository */
+            $repositoryClassName = $this->reflectionService->getClassSchema($className)->getRepositoryClassName();
+            $repository = $this->objectManager->get($repositoryClassName);
```

Comment:
Instead of the inline `@var` I'd go for a real check
```suggestion
            $repository = $this->objectManager->get($repositoryClassName);
            if (!$repository instanceof AssetRepository) {
                throw new \RuntimeException(sprintf('Expected instance of %s, got %s', AssetRepository::class, get_class($repository)), 1575043393);
            }
```
IDEs will pick up on that and we don't risk edge cases

Reply:
Well, the only requirement is "a repository for the class implementing `VariantSupportInterface`", so… will that always be an `AssetRepository`?

Reply:
If we had some interface for repositories of assets implementing `VariantSupportInterface`…